helix-term/src/commands.rs

```diff
@@ -4267,11 +4268,34 @@ fn dap_start(cx: &mut Context) {
     let request = debugger.launch(to_value(args).unwrap());
     let _ = block_on(request).unwrap();
 
-    // TODO: either await "initialized" or buffer commands until event is received
+    let stopped = block_on(debugger.listen_for_event("stopped".to_owned()));
+    let debugger = Arc::new(Mutex::new(debugger));
+    tokio::spawn(dap_listen_stopped(stopped, Arc::clone(&debugger)));
```

Comment:
This is going to race again. We want to handle events in the same way we do on `lsp`: add a `debugger.incoming_events` channel with `enum Event` and receive on it in the main Application loop

Reply:
With those changes there's no need for an Arc<Mutex<>> either since the event loop will synchronize the accesses. I think both this + the change to a `call` method for requests that returns a future we process on `cx.callback` are necessary changes or it will be a messy refactor later on

Reply:
Looks like we need. Thanks, will do that a bit later.